### PR TITLE
Implement a default uncaught exception handler

### LINKED ISSUE #3179: Implement a default uncaught exception handler for REPL
We've been seeing a number of crashes due to errors in libraries and extensions doing work in other threads. We should consider implementing a [global uncaught exception handler](https://developer.android.com/reference/java/lang/Thread#setDefaultUncaughtExceptionHandler(java.lang.Thread.UncaughtExceptionHandler)) in the REPL to process these errors and report them to the user. Currently, these crashes reflect negatively on the app and we may be penalized by Google in the future even though they may originate in an extension.

### PR DESCRIPTION
**Pull request by ewpatton**
Change-Id: Icb5517fad7968978e2a21a4de6078d1407308829

General items:

- [ ] I have updated the relevant documentation files under docs/
- [x] My code follows the:
    - [x] [Google Java style guide](https://google.github.io/styleguide/javaguide.html) (for .java files)
    - [ ] [Google JavaScript style guide](https://google.github.io/styleguide/jsguide.html) (for .js files)
- [x] `ant tests` passes on my machine

If your code changes how something works on the device (i.e., it affects the companion):

- [x] I branched from `ucr`
- [x] My pull request has `ucr` as the base

Further, if you've changed the blocks language or another user-facing designer/blocks API (added a SimpleProperty, etc.):

- [ ] I have updated the corresponding version number in YaVersion.java
- [ ] I have updated the corresponding upgrader in YoungAndroidFormUpgrader.java (components only)
- [ ] I have updated the corresponding entries in versioning.js

For all other changes:

- [ ] I branched from `master`
- [ ] My pull request has `master` as the base

What does this PR accomplish?

*Description*

Fixes #3179 by setting a custom handler for when threads throw uncaught exceptions. Note that the behavior is different between Form and ReplForm, where the latter reports errors back to the editor via RetValManager.

### CODE REVIEW

File: appinventor/components/src/com/google/appinventor/components/runtime/Form.java [hunk 1]
```diff
@@ -311,6 +312,17 @@ public void onCreate(Bundle icicle) {
     // Called when the activity is first created
     super.onCreate(icicle);
 
+    // This version is for production apps. See {@link ReplForm#onCreate} for the REPL version,
+    // which overrides this method.
+    Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
+      @Override
+      public void uncaughtException(@NonNull Thread t, @NonNull Throwable e) {
+        Log.e(LOG_TAG, "Uncaught Exception", e);
+        ErrorOccurred(Form.this, "<unknown>",
```
Comment (jisqyv) [inline appinventor/components/src/com/google/appinventor/components/runtime/Form.java:321]: Doesn't this need to run on the UI thread?

Comment (ewpatton) [reply appinventor/components/src/com/google/appinventor/components/runtime/Form.java:321]: That's a good point. The documentation is unclear where the handler is run so we should play it safe.

**Comment by ewpatton** [comment]
@jisqyv Here is a test app with an extension that just throws an error on another thread using AsynchUtil.

[ExceptionTester.aia.zip](https://github.com/user-attachments/files/17624958/ExceptionTester.aia.zip)

**Review by jisqyv — APPROVED** [review]
LGTM. I'm not sure if it will catch everything, but I believe it will catch most of the things that wind up being reported to the Play Store (and get us in trouble). I tested this by forcing exceptions like the ones I see reported to the Play Store.

**Pull request merged**